Services/OpenIdConnect/classes/class.ilAuthProviderOpenIdConnect.php

```diff
@@ -130,19 +130,24 @@ private function handleUpdate(ilAuthStatus $status, $user_info)
         }
 
         $uid_field = $this->settings->getUidField();
-        $ext_acocunt = $user_info->$uid_field;
+        $ext_account = $user_info->$uid_field;
 
-        $this->getLogger()->debug('Authenticated external account: ' . $ext_acocunt);
+        $this->getLogger()->debug('Authenticated external account: ' . $ext_account);
 
 
         $int_account = ilObjUser::_checkExternalAuthAccount(
             ilOpenIdConnectUserSync::AUTH_MODE,
-            $ext_acocunt
+            $ext_account
         );
 
         try {
             $sync = new ilOpenIdConnectUserSync($this->settings, $user_info);
-            $sync->setExternalAccount($ext_acocunt);
+            if(!is_string($ext_account) || !is_string($int_account)) {
```

Comment:
Are you sure about !is_string($int_account) ? 
In case of a new user (no ILIAS account) tries to authenticate $int_account might be "false". 
Please check this case.

Reply:
Oof, good call. Yeah that led to an "invalid user name or password" which definitely wouldn't have made sense at that point.
Removed it and everything now works as intended, will push changes asap.